jinseok/service/userService.js

```diff
@@ -27,6 +38,50 @@ const signUp = async (
   return createUser;
 };
 
+const login = async (email, password) => {
+  emailValidationCheck(email);
+
+  // get user id and password
+  const userObj = await userDao.getUser(email);
+  const userId = userObj[0].id;
+  const hashedPassword = userObj[0].password;
+
+  // bycrypt verification
+  const checkHash = async (password, hashedPassword) => {
+    return await bcrypt.compare(password, hashedPassword);
+  };
+
+  // check if pw is valid
+  const isPwValid = await checkHash(password, hashedPassword);
+  console.log(isPwValid);
+
+  // check if email is valid
+  const isEmailValid = await userDao.isExistedUser(email);
+
+  let validornot;
+  if (isEmailValid && isPwValid) {
+    try {
+      //토큰 발급
+      const jwtToken = jwt.sign({ id: userId }, process.env.JWT_SECRET);
+
+      validornot = { accessToken: jwtToken };
+    } catch (e) {
+      console.log(e);
+      res.status(400).json({ message: 'error' });
+    }
+    return validornot;
+  }
+  try {
+    validornot = 'Invlaid User';
+  } catch (e) {
+    console.log(e);
+    res.status(400).json({ message: 'error' });
+  }
+
+  return validornot;
+};
```

Comment:
```suggestion
const login = async (email, password) => {
  await emailValidationCheck(email);
  
  const [user] = await userDao.getUser(email);
  
  if(!user || !bcrypt.compare(password, user.password)) throw new Error("INVALID_EMAIL_OR_PASSWORD");
}

  return jwt.sign({ id : user.id }, process.env.JWT_SECRET);
```
전체적으로 로직이 불필요하게 복잡한 것 같습니다! try catch도 꼭 필요한 부분에서만 사용하셔도 되고, 이미 해당 service 함수를 사용하고 있는 controller에서 try catch를 작성하였기 때문에 필요 외에는 작성하지 않아도 됩니다. 또한 예외 처리로 response를 보내고 있는데, 이는 layered architecture가 적용된 코드에서 올바르지 않은 코드입니다! controller에서 처리할 수 있도록 해주시고 service에는 res 객체가 존재하지 않기 때문에 원할히 동작하지 않을 수 있습니다.